src/url-replacements.js

```diff
@@ -144,6 +148,82 @@ class UrlReplacements {
     this.set_('SCREEN_HEIGHT', () => {
       return this.win_.screen.height;
     });
+
+    // Returns the time it took to load the whole page. (excludes amp-* elements
+    // that are not rendered by the system yet.)
+    this.set_('PAGE_LOAD_TIME', () => {
+      return this.getTimingData_('navigationStart', 'loadEventStart');
+    });
+
+    // Returns the time it took to perform DNS lookup for the domain.
+    this.set_('DOMAIN_LOOKUP_TIME', () => {
+      return this.getTimingData_('domainLookupStart', 'domainLookupEnd');
+    });
+
+    // Returns the time it took to connet to the server.
+    this.set_('TCP_CONNECT_TIME', () => {
+      return this.getTimingData_('connectStart', 'connectEnd');
+    });
+
+    // Returns the time it took for server to start sending a response to the
+    // request.
+    this.set_('SERVER_RESPONSE_TIME', () => {
+      return this.getTimingData_('requestStart', 'responseStart');
+    });
+
+    // Returns the time it took to download the page.
+    this.set_('PAGE_DOWNLOAD_TIME', () => {
+      return this.getTimingData_('responseStart', 'responseEnd');
+    });
+
+    // Returns the time it took for redirects to complete.
+    this.set_('REDIRECT_TIME', () => {
+      return this.getTimingData_('navigationStart', 'fetchStart');
+    });
+
+    // Returns the time it took for DOM to become interactive.
+    this.set_('DOM_INTERACTIVE_TIME', () => {
+      return this.getTimingData_('navigationStart', 'domInteractive');
+    });
+
+    // Returns the time it took for content to load.
+    this.set_('CONTENT_LOAD_TIME', () => {
+      return this.getTimingData_('navigationStart',
+          'domContentLoadedEventStart');
+    });
+  }
+
+  /**
+   * Returns navigation timing information based on the start and end events.
+   * The data for the timing events is retrieved from performance.timing API.
+   * @param {string} startEvent
+   * @param {string} endEvent
+   * @return {!Promise<string>}
+   * @private
+   */
+  getTimingData_(startEvent, endEvent) {
+    const timingInfo = this.win_['performance']
+        && this.win_['performance']['timing'];
+    if (!timingInfo || timingInfo['navigationStart'] == 0) {
+      // Navigation timing API is not supported.
+      return Promise.resolve();
+    }
+
+    let metric = timingInfo[endEvent] - timingInfo[startEvent];
+    if (isNaN(metric) || metric == Infinity) {
+      // The metric is not supported.
+      return Promise.resolve();
+    } else if (metric < 0) {
+      // Metric is not yet available. Retry after a delay.
+      return timer.promise(TIMING_API_CALCULATION_DELAY).then(() => {
```

Comment:
Don't think so: https://developer.mozilla.org/en-US/docs/Web/API/Window/performance.

I could wait for page load event and then set the values but that would mean that by the time all the values are set, request might not get sent. Would you suggest that?  (this happens frequently with bounces.)